src/System.Private.Uri/src/System/DomainNameHelper.cs

```diff
@@ -324,7 +324,7 @@ internal unsafe static string IdnEquivalent(char* hostname, int start, int end,
             {
                 // just lowercase for ascii
                 string unescapedHostname = new string(hostname, start, end - start);
-                return ((unescapedHostname != null) ? unescapedHostname.ToLowerInvariant() : null);
+                return unescapedHostname.ToLowerInvariant();
```

Comment:
I was going to suggest that we should have an Assert about not being null.  But maybe the "coverity" tool (I'm not that familiar with it) is sufficient for static analysis.